frontend/main.js

```diff
@@ -1,13 +1,74 @@
 const { app, BrowserWindow, dialog, ipcMain } = require("electron/main");
 const path = require("node:path");
-
+const fs = require("fs-extra");
+const { PythonShell } = require("python-shell");
+const constants = require("./constants/constants.js");
 let mainWindow;
 
+function runProgram(filePath) {
+  mainWindow.loadFile("loading.html");
+
+  const script = constants.SCRIPT;
+
+  const poetryArgs = [
+    constants.COMMANDS.DEDUPE_DATA,
+    constants.OPTIONS.FORMAT,
+    constants.FORMAT.TEST,
+    filePath,
+    "./",
+  ];
+
+  let options = {
+    mode: "text",
+    scriptPath: "../cli",
+    pythonPath: "../.venv/bin/python",
```

Comment:
Also, is the `.venv` inclusion here just needed for development or will that be necessary when we've bundled with Electron? It's fine for now, but we might want to make that environment-specific

Reply:
Yeah this is just needed for development. Gotcha, I will make the change once I start on packaging/distribution